src/lib/tls/asio/asio_stream.h

```diff
@@ -676,25 +721,72 @@ class Stream
                it++)
             {
             const boost::asio::const_buffer buffer = *it;
-            try
+            try_with_error_code([&]
                {
                native_handle()->send(static_cast<const uint8_t*>(buffer.data()), buffer.size());
-               }
-            catch(const TLS_Exception& e)
-               {
-               ec = e.type();
-               }
-            catch(const Botan::Exception& e)
-               {
-               ec = e.error_type();
-               }
-            catch(const std::exception&)
-               {
-               ec = Botan::ErrorType::Unknown;
-               }
+               }, ec);
+            }
+         }
+
+      /**
+       * @brief Map an EOF error_code returned by the underlying transport according to state of the SSL session.
+       *
+       * Returns a const reference to the error code object, suitable for passing to a completion handler.
+       * This method corresponds to the method map_error_code in Boost.Asio's SSL engine.
+       */
+      const boost::system::error_code map_error_code(boost::system::error_code& ec)
+         {
+         // We only want to map the error::eof code.
+         if(ec != boost::asio::error::eof)
+            {
+            return ec;
+            }
+
+         // TODO
```

Comment:
I am very confused by this one. @reneme @Tymolc let's talk about it in the chat.